internal/templates/weld.tmpl

```diff
@@ -23,6 +23,16 @@ func Make{{.BackendsName}}({{.ParamsString}}) ({{.BackendsType}}, error) {
 {{if .Provider.ReturnsErr}}if err != nil { return nil, errors.Wrap(err, "{{.Provider.ErrWrapMsg}}") }{{end}}
 {{end}}{{end}}
 
+{{if .Overrides  }}
```

Comment:
It looks like this will end up generating a different signature for the makeBackends which then is used by production code. I think we can avoid this.

Can we wrap the whole Backends in another struct which is only used for testing. Something like
```
type Overrideable struct {
   Backends
   overrideDB DB
}

func (o Overrideable) GetDB() {
   if o.overrideDB != nil {
       return o.overrideDB
   }
   return o.Backends.GetDB()
}

func OverrideBackendsForTesting(t *testing.T, opts ...Option) {
   ...
```
this can allow you to create the backends in a normal way or maybe a stub one which isn't initialised and then you can use this function to wrap it and override dependencies.

Reply:
It wont be as there will be a seperate generated file for testing. You would land up creating a "testing state" that runs with -override. The production code would not use that.

Example of layout could be:

state/
  weld_gen.go [prod state]
  devstate/
    weld_gen.go [dev state]
  teststate/
    weld_gen.go [testing state]

Reply:
I still think it would be cleaner if the override flag output a function next to the devstate (assuming that's the default for testing). Rather than making the weld template larger and more complex, we could have an override template that is completely separate and only generated when the flag is passed in.

Reply:
Okay will look at setting aside the override template

Reply:
Updated the MR to keep the original template unchanged